packages/dds/matrix/src/sparsearray2d.ts

```diff
@@ -219,17 +235,19 @@ export class SparseArray2D<T>
 		}
 	}
 
-	private getLevel<T>(parent: UA<UA<T>>, subKey: number) {
+	private getLevel<T>(parent: UA<UA<T>>, subKey: number): UA<T> {
 		const level = parent[subKey];
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
-		return level ?? (parent[subKey] = new Array(256).fill(undefined));
+		// Using new Array is needed because the array created with Array.from does not
+		// satisfy (T|undefined)[].
+		// eslint-disable-next-line unicorn/no-new-array
+		return level ?? (parent[subKey] = new Array<T|undefined>(256).fill(undefined));
 	}
 
-	public snapshot() {
+	public snapshot(): UA<UA<UA<UA<UA<T>>>>> {
```

Comment:
Whoa. This is funky. I don't know this code well, but just to sanity check: is this implementing an interface method with simpler signature?

Reply:
No, unfortunately not :(. It follows the typing of the root, which was defined ~4 years ago (presumably when the file was created).

Reply:
OOF. Okay, well if nothing else, this maybe exposes some previously hidden code smell.